### PR TITLE
Use native cache from actions/setup-go

### PR DESCRIPTION
**Pull request by mmorel-35**
Thank you for contributing to Velero!

# Please add a summary of your change

Use native cache from actions/setup-go instead of defining actions/cache for `~/go/pkg/mod` path

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: .github/workflows/push.yml
Comment (kaovilai) [inline .github/workflows/push.yml:1]: This file only have formatting changes right? Which part is `Use native cache from actions/setup-go`?

**Comment by reasonerjt** [comment]
Doesn't have to be in v1.14.0?  I'd propose we delay it after v1.14.0 RC

File: .github/workflows/push.yml
Comment (kaovilai) [reply .github/workflows/push.yml:1]: ok cool. Hiding whitespace clears this up.

**Review by kaovilai — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Comment by blackpiglet** [comment]
@ywk253100 @reasonerjt 
Please take a look. Ready for review.

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -14,95 +14,82 @@ jobs:
...
-        sudo swapoff -a
-        sudo rm -f /mnt/swapfile
-        docker system prune -a --force
+      - name: Check out the code
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+        id: go
```
Comment (mateusoliveira43) [inline .github/workflows/push.yml:23]: I would remove `id`s

**Review by blackpiglet — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**